### PR TITLE
Aegis 222 new

### PR DESCRIPTION
**Pull request by TanujaG**
@loahou04 @richweclg @upopeap @marcdonovan @vshivp2 Sorry for multiple pull request... bugs for 222,221,225,231

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -70,7 +72,7 @@ class NotificationComponent {
 			recipientId: message.payload.recipientId
 		});
 
-		this._sortNotificationList();
+		this._sortList('NOTIFICATION-LIST');
```
Comment (loahou04) [inline main.js]: Ok i set this up wrong and you are just following my pattern.  The right way to do this is to go ahead and set the createdAt and updatedAt to Date objects inside the api return cal.  Then we don't need a function to do this for us and when we want to sort the array we just call list.sort((x, y)=>{return y.createdAt - x.createdAt;})

**Comment by richweclg** [comment]
approved

File: src/js/NotificationContainer.js [hunk 2]
```diff
@@ -85,7 +94,7 @@ export default class NotificationContainer extends React.Component {
 					</h1>
 				</div>
 				<div className="notification-archive--back">
-					<button onClick={this.state.isArchive && !this.state.displayDetails ? this.showNonArchivedList.bind(this) : this.props.closeDrawer}> <i className={this.state.isArchive && !this.state.displayDetails ? 'pe-icon--chevron-down pointer' : 'pe-icon--times close-dropdown pointer'}></i> </button>
+					<button onClick={this.state.isArchive && !this.state.displayDetails ? this.showNonArchivedList.bind(this) : this.resetListOnCloseDrawer.bind(this)}> <i className={this.state.isArchive && !this.state.displayDetails ? 'pe-icon--chevron-down pointer' : 'pe-icon--times close-dropdown pointer'}></i> </button>
```
Comment (loahou04) [inline src/js/NotificationContainer.js]: something i literally learned yesterday...these bind calls are expensive.  Lets move them to the constructor

File: main.js [hunk 3]
```diff
@@ -41,7 +42,12 @@ class NotificationComponent {
 			this.newNotifications = result.newNotifications;
 			this.unreadCount = result.unreadCount;
 
-			this._sortNotificationList();
+			this.notificationList.sort((x, y) => {
```
Comment (loahou04) [inline main.js]: This is nit picky...but marc would call you out on this as well.  DRY....don't repeat yourself.  Extract it out to a function and just pass the function in.

**Comment by loahou04** [comment]
approved

**Pull request merged**